lib/rules/comment-pattern/__tests__/index.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const { messages, ruleName } = require('..');
+
+testRule({
+	ruleName,
+	config: [/foo-.+/],
```

Comment:
Could you also make a copy of this `testRule`, but use `'foo-.+'` string instead of RegEx. To make sure both RegEx and string option work exactly the same.

Reply:
Instead of copy-pasting the entire call for `testRule`, what do you think about iterating the configurations that should behave in the exact same way, and calling `testRule` for each?
Meaning:
```js
[/foo-.+/, 'foo-.+'].forEach((config) => {
	testRule({
		ruleName,
		config: [config],
                 ...
```

This way, it is clear that the two configurations behave exactly the same, and if for some reason the tests will require updates - it will be in a single place.
The downside that it could be less readable when seeing for the first time (I didn't find something similar in the existing tests)